libcodechecker/analyze/log_parser.py

```diff
@@ -196,12 +221,11 @@ def parse_compile_commands_json(logfile, add_compiler_defaults=False):
                         if re.match(pattern, comp_opt):
                             extra_opts.append(comp_opt)
 
-                compiler_defines[results.compiler] = \
-                    get_compiler_defines(results.compiler, extra_opts)
                 compiler_includes[results.compiler] = \
                     get_compiler_includes(results.compiler, extra_opts)
-            action.compiler_defines = compiler_defines[results.compiler]
+                compiler_target = get_compiler_target(results.compiler)
             action.compiler_includes = compiler_includes[results.compiler]
+            action.analyzer_options.append("--target=" + compiler_target)
```

Comment:
Let's introduce an empty newline before this function to make this code better readable.